app/member-reminders/member-reminders-public/src/main/kotlin/com/hedvig/android/memberreminders/GetConnectPaymentReminderUseCase.kt

```diff
@@ -8,28 +8,37 @@ import com.hedvig.android.apollo.safeExecute
 import com.hedvig.android.apollo.toEither
 import com.hedvig.android.core.common.ErrorMessage
 import com.hedvig.android.logger.logcat
+import kotlinx.datetime.LocalDate
 import octopus.GetPayinMethodStatusQuery
 import octopus.type.MemberPaymentConnectionStatus
 
 internal interface GetConnectPaymentReminderUseCase {
-  suspend fun invoke(): Either<ConnectPaymentReminderError, ShowConnectPaymentReminder>
+  suspend fun invoke(): Either<ConnectPaymentReminderError, PaymentReminder>
 }
 
 internal class GetConnectPaymentReminderUseCaseImpl(
   private val apolloClient: ApolloClient,
 ) : GetConnectPaymentReminderUseCase {
-  override suspend fun invoke(): Either<ConnectPaymentReminderError, ShowConnectPaymentReminder> {
+  override suspend fun invoke(): Either<ConnectPaymentReminderError, PaymentReminder> {
     return either {
-      val payinStatus = apolloClient.query(GetPayinMethodStatusQuery())
+      val result = apolloClient.query(GetPayinMethodStatusQuery())
         .safeExecute()
         .toEither(::ErrorMessage)
         .mapLeft(ConnectPaymentReminderError::NetworkError)
         .bind()
-
-      ensure(payinStatus.currentMember.paymentInformation.status == MemberPaymentConnectionStatus.NEEDS_SETUP) {
-        ConnectPaymentReminderError.AlreadySetup
+      val missingPaymentsContractTerminationDate = result.currentMember.activeContracts
+        .filter { it.terminationDueToMissedPayments }
+        .sortedBy { it.terminationDate }
+        .firstOrNull()?.terminationDate
+      if (missingPaymentsContractTerminationDate != null) {
+        PaymentReminder.ShowMissingPaymentsReminder(missingPaymentsContractTerminationDate)
+      } else {
+        val payStatus = result.currentMember.paymentInformation.status
+        ensure(payStatus == MemberPaymentConnectionStatus.NEEDS_SETUP) {
+          ConnectPaymentReminderError.AlreadySetup
+        }
+        PaymentReminder.ShowConnectPaymentReminder
       }
```

Comment:
```suggestion
      val missingPaymentsContractTerminationDate = result.currentMember.activeContracts
        .filter { it.terminationDueToMissedPayments }
        .sortedBy { it.terminationDate }
        .firstOrNull()?.terminationDate
      if (missingPaymentsContractTerminationDate != null) {
        return PaymentReminder.ShowMissingPaymentsReminder(missingPaymentsContractTerminationDate)
      }
      val payStatus = result.currentMember.paymentInformation.status
      ensure(payStatus == MemberPaymentConnectionStatus.NEEDS_SETUP) {
        ConnectPaymentReminderError.AlreadySetup
      }
      PaymentReminder.ShowConnectPaymentReminder
```
I think this can look a tiny bit simpler with one less indentation if we just return early in the scenario where we know for sure we want to show that type of reminder

Reply:
It may be `return@either` instead, not sure try to see what the IDE tells you